### PR TITLE
2.next Add DboSource::flushQueryCache()

### PR DESCRIPTION
**Pull request by chinpei215**
Seems to be no way to flush query caches.

### CODE REVIEW

File: lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php [hunk 1]
```diff
@@ -78,6 +78,14 @@ public function nestedTransactionSupported() {
 		return $this->useNestedTransactions && $this->nestedSupport;
 	}
 
+	public function writeQueryCache($sql, $data, $params = array()) {
+		return $this->_writeQueryCache($sql, $data, $params);
+	}
```
Comment (markstory) [inline lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php]: Instead of an testing hook method, could we just execute SQL?

Comment (chinpei215) [inline lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php]: @markstory Can we use `ReflectionProperty::setAccessible()` instead? As `_queryCache` is a protected property, we cannot make sure whether it being flushed or not in the test.

Comment (markstory) [inline lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php]: You can use `assertAttributeEqual()` to test the value of protected properties.

Comment (chinpei215) [inline lib/Cake/Test/Case/Model/Datasource/DboSourceTest.php]: @markstory Thank you very much! I didn't know the method.

**Comment by chinpei215** [comment]
Done. I've used `assertAttributeCount()` to count a protected array.

**Comment by markstory** [comment]
Awesome work @chinpei215 :+1:

**Pull request merged**